src/App.js

```diff
@@ -14,11 +16,96 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    visibility: false,
+    goods: [...goodsFromServer],
+    selected: 1,
+  }
+
+  start = () => {
+    this.setState(
+      {
+        visibility: true,
+      },
+    );
+  }
+
+  reverse = () => {
+    this.setState(prev => (
+      {
+        goods: [...prev.goods.reverse()],
+      }
+    ));
+  }
+
+  sort = () => {
+    this.setState(prev => (
+      {
+        goods: [...prev.goods.sort()],
+      }
+    ));
+  }
+
+  sortLength = () => {
+    this.setState(prev => (
+      {
+        goods: [...prev.goods.sort((a, b) => (a.length - b.length))],
+      }
+    ));
+  }
+
+  reset = () => {
+    this.setState(() => (
+      {
+        goods: [...goodsFromServer],
+        selected: 1,
+      }
+    ));
+  }
+
+  selectLength = (event) => {
+    const length = Number(event.target.value);
+
+    this.setState(() => ({
+      goods: [...goodsFromServer.filter(item => (
+        item.length >= length
+      ))],
+      selected: length,
+    }));
```

Comment:
Тут я теж не одразу сприйняв
```suggestion
    this.setState(() => ({
      goods: [
        ...goodsFromServer
          .filter(item => (item.length >= length))
      ],
      selected: length,
    }));
```
Хоча може є ще кращий спосіб подачі

Reply:
fixed